ote/src/clj/ote/transit_changes/detection.clj

```diff
@@ -193,14 +189,12 @@
     :default
     state))
 
-(defn week-hash-no-traffic-run
-  "Return the continuous amount of days that have no traffic at the beginning or end of the weekhash."
-  [beginning? weekhash]
-  (->> (if beginning?
-         weekhash
-         (reverse weekhash))
-       (take-while #(or (nil? %) (keyword? %)))
-       count))
+(defn week-hash-key-ix
+  "Input: weekhash = sequence of string hashes
+    key-to-find = key to find
+  Output: Returns the index of first occurrence of `key-to-find`. Monday = 0, Sunday = 6, nil = not found"
+  [weekhash key-to-find]
+  (key-to-find (zipmap weekhash (range 0 8))))
```

Comment:
(range 8) is the same as (range 0 8)

Reply:
True, intentionally used this for clarity, but can remove.